Swetrix/swetrix
# (feature) webhook alerts

Please describe your changes using the checklist below.

### Self-hosted support
- [ ] Your feature is implemented for the selfhosted version of Swetrix
- [x] This PR only updates the production code (e.g. paddle webhooks, CAPTCHA, blog, etc.)

### Database migrations
- [x] Clickhouse / MySQL migrations added for this PR
- [ ] No table schemas changed in this PR

### Documentation
- [ ] You have updated the [documentation](https://github.com/swetrix/docs) according to your PR
- [x] This PR did not change any publicly documented endpoints



How to test this feature:

- Insert the following data into the db
```
mysql> select * from users_webhooks;
+--------------------------------------+--------------------------------------+------+-------------------------------+----------------------------+----------------------------+
| id                                   | userId                               | name | url                           | createdAt                  | updatedAt                  |
+--------------------------------------+--------------------------------------+------+-------------------------------+----------------------------+----------------------------+
| f04816b4-7516-44ba-9fb3-82e065667785 | c6df6997-80a9-4b27-9248-a15af0b97ca9 | Test | http://localhost:3000/webhook | 2024-07-11 21:12:51.266892 | 2024-07-12 20:05:35.460372 |
+--------------------------------------+--------------------------------------+------+-------------------------------+----------------------------+----------------------------+
```
- Create express JS server

```javascript
const express = require('express');
const bodyParser = require('body-parser');
const app = express();
const port = 3000;

app.use(bodyParser.json());

app.post('/webhook', (req, res) => {
    console.log('Received webhook:', req.body);

    res.status(200).send('Webhook received');
});

app.listen(port, () => {
    console.log(`Webhook receiver listening at http://localhost:${port}`);
});

```

You should see the following output in the console:

```
 {
  event: 'user.alert.metrics',
  data: {
    alertName: 'Test Alert',
    projectName: 'Your awesome project',
    count: 0,
    queryMetric: 'page_views',
    queryCustomEvent: null,
    queryTime: '0'
  }
}
```
